lib/src/test/java/io/ably/lib/test/rest/RestAuthTest.java

```diff
@@ -79,6 +79,23 @@ public void authinit0() {
 	}
 
 	/**
+	 * Init library with useTokenAuth set
+	 */
+	@Test
+	public void authinit0point5() {
+		try {
+			TestVars testVars = Setup.getTestVars();
+			ClientOptions opts = new ClientOptions(testVars.keys[0].keyStr);
+			opts.useTokenAuth = true;
+			AblyRest ably = new AblyRest(opts);
+			assertEquals("Unexpected Auth method mismatch", ably.auth.getAuthMethod(), AuthMethod.token);
```

Comment:
Good plan. I've just done a realtime test for it, so I think I'll do what you suggest and have both.